src/queryResult/queryResultWebViewController.ts

```diff
@@ -60,17 +72,105 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
             state.tabStates.resultPaneTab = payload.tabId;
             return state;
         });
+        this.registerReducer("getExecutionPlan", async (state, payload) => {
+            this._executionPlanContents = payload.sqlPlanContent;
+            await this.createExecutionPlanGraphs();
+            state.loadState = ApiStatus.Loaded;
+            state.tabStates.resultPaneTab =
+                qr.QueryResultPaneTabs.ExecutionPlan;
+            return {
+                ...state,
+                sqlPlanContent: this._executionPlanContents,
+                executionPlan: this.state.executionPlan,
+                executionPlanGraphs: this.state.executionPlanGraphs,
+            };
+        });
+        this.registerReducer("saveExecutionPlan", async (state, payload) => {
+            let folder = vscode.Uri.file(homedir());
+            if (await exists("Documents", folder)) {
+                folder = vscode.Uri.file(path.join(folder.path, "Documents"));
+            }
+
+            let filename: vscode.Uri;
+            let counter = 1;
+            if (await exists(`plan.sqlplan`, folder)) {
+                while (await exists(`plan${counter}.sqlplan`, folder)) {
+                    counter += 1;
+                }
+                filename = vscode.Uri.joinPath(
+                    folder,
+                    `plan${counter}.sqlplan`,
+                );
+            } else {
+                filename = vscode.Uri.joinPath(folder, "plan.sqlplan");
+            }
+
+            // Show a save dialog to the user
+            const saveUri = await vscode.window.showSaveDialog({
+                defaultUri: filename,
+                filters: {
+                    "SQL Plan Files": ["sqlplan"],
+                },
+            });
+
+            if (saveUri) {
+                // Write the content to the new file
+                await vscode.workspace.fs.writeFile(
+                    saveUri,
+                    Buffer.from(payload.sqlPlanContent),
+                );
+            }
+
+            return state;
+        });
+        this.registerReducer("showPlanXml", async (state, payload) => {
+            const planXmlDoc = await vscode.workspace.openTextDocument({
+                content: payload.sqlPlanContent,
+                language: "xml",
+            });
+
+            await vscode.window.showTextDocument(planXmlDoc);
+
+            return state;
+        });
+        this.registerReducer("showQuery", async (state, payload) => {
+            await this.untitledSqlDocumentService.newQuery(payload.query);
+
+            return state;
+        });
+        this.registerReducer("updateTotalCost", async (state, payload) => {
+            this.state.totalCost += payload.totalCost;
```

Comment:
The arg is called `totalCost`, which makes it sound like this should be an assignment instead of an addition.  Should this be `this.state.totalCost = payload.totalCost` or `this.state.totalCost += addedCost`?